pkgs/development/go-modules/generic/default.nix

```diff
@@ -198,7 +198,7 @@ go.stdenv.mkDerivation (
      ln -s "${dep.src}" "$d/src/${dep.goPackagePath}"
   ''
   ) goPath) + ''
-    export GOPATH="$d:$GOPATH"
+    export GOPATH=${lib.concatStringsSep ":" ( ["$d"] ++ ["$GOPATH"] ++ ["$PWD"] ++ extraSrcPaths)}
```

Comment:
Let's do it. You can't break Go dependency resolution more :/
It will have even more *corner cases* with Go 1.8 https://beta.golang.org/doc/go1.8#gopath